src/cmds/qstat.c

```diff
@@ -302,49 +359,174 @@ states(char *string, char *q, char *r, char *h, char *w, char *t, char *e, int l
  * @param[in] n - attribute name
  * @param[in] r - resource name
  * @param[in] v - value for the attribute
+ * @param[in] one_line - one line per attribute
  *
  * @return Void
  *
  */
 void
-prt_attr(char *n, char *r, char *v)
+prt_attr(char *n, char *r, char *v, int one_line)
 {
-	char *c;
-	char *comma = ",";
-	int   first = 1;
-	int   l;
-	int   start;
-
-	start = strlen(n) + 7;	/* 4 spaces + ' = ' is 7 */
-	printf("    %s", n);
-	if (r) {
-		start += strlen(r) + 1;
-		printf(".%s", r);
-	}
-	printf(" = ");
-	c = strtok(v, comma);
-	while (c) {
-		if ((l = strlen(c)) + start < 78) {
-			printf("%s", c);
-			start += l;
-		} else {
-			if (! first) {
-				printf("\n\t");
-				start = 9;
+	int            first = 1;
+	int            l = 0;
+	int            start = 0;
+	int            i = 0;
+	char           *c = NULL;
+	char           *comma = ",";
+	static char    *prev_resc_name = NULL;
+	char           *buf = NULL;
+	char           *keyvalpair = NULL;
+	char           *key = NULL;
+	char           *val = NULL;
+	char           *temp = NULL;
+
+	if(v == NULL)
+		return;
+	switch (output_format) {
+		case FORMAT_JSON:
+			if (strcmp(n, ATTR_v) == 0) {
+				if (add_json_node(JSON_OBJECT, JSON_NULL, n, NULL) == NULL) {
+					fprintf(stderr, "qstat: out of memory\n");
+					exit(1);
+				}
+				keyvalpair = (char *) malloc(MAXPATHLEN);
+				l = MAXPATHLEN;
```

Comment:
how is my comment making you think that my suggestion will be restricting the user from using long values? Please clarify.
My understanding is that 
'v' will contain something like --> "a=123,bcd=45678,xyz=91230, ....".
keyvalpair will contain --> "a=123" then "bcd=45678", then "xyz=91230" and so on...
If the value contains an escaped comma, you are removing the escape character.
So, at any given time, the length of keyvalpair will not extend the length of 'v' (it will be same only in the case where you have only one variable in 'v').
So, if you allocate strlen(v) bytes to start with you can do away with the realloc logic below.
Hope this is clear to you.

Regarding the variable names being legacy code - You have changed a lot of legacy code already, don't feel sorry to change the names of variables if they make reading of the code better...